pallets/creditcoin/src/tests.rs

```diff
@@ -341,19 +376,60 @@ fn register_address_should_work() {
 }
 
 #[test]
-fn register_address_pre_existing() {
+fn register_address_should_fail_to_reregister_external_address_to_same_account() {
 	ExtBuilder::default().build_and_execute(|| {
-		let (who, address, ownership_proof, _) = generate_address_with_proof("owner");
+		let (who, _) = generate_random_account("owner");
+		let (external_address, external_keypair) =
+			generate_random_external_address("external owner");
+		let ownership_proof = build_proof_of_ownership(who.clone(), external_keypair);
```

Comment:
Q: why separate account, address and ownership proof generation into multiple functions? 

From what I can tell that isn't necessary for this test scenario. Am I missing something here?

Reply:
You're 100% correct. I just separated them out for my own sanity/understanding.

Reply:
I changed it back in the spirit of changing few things.